### ISSUE TITLE
Profiles have URI identifiers that resolve to more detailed descriptions (6.1)

### ISSUE THREAD

**Issue by kcoyle** [w3c/dxwg#205]
#75

**Comment by andrea-perego** [comment]
If the distinction I made at http://lists.w3.org/Archives/Public/public-dxwg-wg/2018May/0144.html between "profile description" and "profile definition(s)" makes sense, I would say that the URI should return the "profile description" (i.e., profile metadata, possibly in different formats - e.g., RDF & HTML), which includes links to the available profile definitions.

**Comment by kcoyle** [comment]
Andrea, although this has its own logic, this does not meet the use case provided for content negotiation. I think we need to hear from Ruben and Lars on this as it would seem to require an additional interaction beyond http. See https://w3c.github.io/dxwg/ucr/#ID2

**Comment by andrea-perego** [comment]
Thanks, @kcoyle . So looking at [the relevant UC](https://w3c.github.io/dxwg/ucr/#ID2), I guess that "the more detailed descriptions" are specifically referring to the HTTP response headers telling which are the available profiles and, for each of them, which formats are available.

The two things are not mutually exclusive: the URI can return both such HTTP headers and the profile metadata. And actually this is approach used in the [GeoDCAT-AP API](https://webgate.ec.europa.eu/CITnet/stash/projects/ODCKAN/repos/iso-19139-to-dcat-ap/browse/api) - quoting:

<blockquote>
<p>Besides the resulting RDF serialisation of the source ISO 19139 records, the API returns a set of HTTP <a href="https://tools.ietf.org/html/rfc5988"><code>Link</code></a> headers, and the corresponding HTML LINK elements in the HTML+RDFa serialisation.</p>

<table width="100%">
  <thead>
    <tr>
      <th><a href="http://www.iana.org/assignments/link-relations/" title="IANA Link Relations">Relation type</a></th>
      <th>Type</th>
      <th>Title</th>
      <th>Target URI</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code>derivedfrom</code></td>
      <td><code>application/xml</code></td>
      <td><code>ISO 19139</code></td>
      <td>The URL of the source document, containing the ISO 19139 records.</td>
    </tr>
    <tr>
      <td rowspan="2"><code>profile</code></td>
      <td rowspan="2">The media type of the document returned by the API.</td>
      <td><code>DCAT-AP</code></td>
      <td><code>core</code></td>
    </tr>
    <tr>
      <td><code>GeoDCAT-AP</code></td>
      <td><code>extended</code></td>
    </tr>
    <tr>
      <td><code>self</code></td>
      <td>The media type of the document returned by the API.</td>
      <td>The name of the returned RDF serialisation.</td>
      <td>The URL of the document returned by the API.</td>
    </tr>
    <tr>
      <td><code>alternate</code></td>
      <td>The media types of the alternative RDF serialisations supported by the API.</td>
      <td>The name of the relevant RDF serialisation.</td>
      <td>The URL of the document, encoded with the relevant RDF serialisation, as would be returned by the API.</td>
    </tr>
  </tbody>
</table>
</blockquote>

**Comment by kcoyle** [comment]
Thanks, Andrea. This is very helpful. Now I hope that Lars & Ruben weigh in as to whether this fits with their concept of the requirements on servers in response to the headers they envision.

**Comment by larsgsvensson** [comment]
Only speaking for myself here (and hoping that @RubenVerborgh weighs in if he has a different view on this), the use of `Link` headers (and html `link` elements) to point clients to other representations definitely fits with the concept. My preferred solution would be a `profile` relation type (i. e. not `rel="profile"`but `profile="https://example.org/profiles/1"`) just as @andrea-perego lists in his example (although I guess that there's a copy-and-paste error and that the profile is not about the media type...).
So I think we're on the same page here.

**Comment by RubenVerborgh** [comment]
+1

**Comment by larsgsvensson** [comment]
Great, then we're on the same page

**Comment by aisaac** [comment]
I have an additional question: I agree that for a basic request for an 'holistic' data container like RDF, the URI of the profile should return to the 'hub' description in profileDesc. But what if the request is for XML? And could it be that for a request for RDF, asking for a 'SHACL' profile (ouch, metaprofiling ahead!!!) then what would get returned would be an XML Schema, or a SHACL file?

**Comment by RubenVerborgh** [comment]
> (ouch, metaprofiling ahead!!!)

You said that like it's a bad thing :-)
It's perfect that we can reuse the same mechanism on profiles too.

**Comment by andrea-perego** [comment]
I agree with @RubenVerborgh . A profile is just another resource, and shouldn't be treated differently.

I'm probably missing some background discussion, but I'm not clear whether the current agreement is that a profile URI MUST or instead SHOULD be dereferenceable. I would be in favour of SHOULD, and to use MUST only for requiring a profile to have a URI (irrespective of whether it is dereferenceable or not).

**Comment by kcoyle** [comment]
+1 Lars, Ruben, Andrea. In terms of "SHOULD be dereferenceable", another way is to couch it in terms of functionality: if you want to do X then the profile MUST be dereferenceable; if it isn't dereferenceable, then you cannot do X. People should be able to choose what services and what functions they want to support, and to understand what they need to provide to get the services they desire.

**Comment by andrea-perego** [comment]
@kcoyle said:

> +1 Lars, Ruben, Andrea. In terms of "SHOULD be dereferenceable", another way is to couch it in terms of functionality: if you want to do X then the profile MUST be dereferenceable; if it isn't dereferenceable, then you cannot do X. People should be able to choose what services and what functions they want to support, and to understand what they need to provide to get the services they desire.

+10! (factorial :)

**Comment by aisaac** [comment]
ok it seems we do have an agreement here!

Now should this requirement be added to the ones on our requirement working space at https://docs.google.com/document/d/13hV2tJ6Kg2Hfe7e1BowY5QfCIweH9GxSCFQV1aWtOPg ? Or perhaps we need to attach it to a use case before. I'm not yet sure we have one, though.

**Comment by kcoyle** [comment]
I'm not sure we've resolved this one completely. We have a requirement that profiles have identifiers, but the discussion here seems to be about what those identifiers resolve to. Can someone identify the relevant use case and find or create a requirement that fits this discussion? Thanks!

**Comment by rob-metalinkage** [comment]
This issue is addressed by the use of URIs in conneg-by-ap and the provision of the profiles vocabulary and accepted requirements in the prof-guidance doc.

**Comment by kcoyle** [comment]
Does this mean that we do not need to address this in profile guidance? 

I ask because it seems that this fits in to describing profiles as being sets of resources, and it would be necessary for the members of those sets to be available/linked from the profile identifier. Without this I don't know how we will have a platform to talk about the profile vocabulary, as this would be the motivation to include that in the guidance document.
